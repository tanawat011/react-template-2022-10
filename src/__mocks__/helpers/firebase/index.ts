export const deleteCollection = jest.fn()
export const deleteDocument = jest.fn()
export const getDocument = jest.fn()
export const getAllDocuments = jest.fn()
export const setDocument = jest.fn()
export const subscribeCollection = jest.fn()
export const updateDocument = jest.fn()
