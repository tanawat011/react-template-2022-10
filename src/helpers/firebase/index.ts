export * from './deleteCollection'
export * from './deleteDocument'
export * from './getAllDocuments'
export * from './getDocument'
export * from './setDocument'
export * from './subscribeCollection'
export * from './updateDocument'
