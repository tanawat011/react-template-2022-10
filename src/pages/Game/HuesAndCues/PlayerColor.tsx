import type { RoomPlayer } from './_type'

import { TwRow } from 'components/Common'

import { CellClickable, MarkerLG } from './Common'

type Prop = {
  currentPlayer: RoomPlayer
  playersInRoom: RoomPlayer[]
  updateCurrentPlayer: (currentPlayer: RoomPlayer) => void
}

export const PlayerColor: React.FC<Prop> = ({ currentPlayer, playersInRoom }) => {
  return (
    <div className='mb-2 p-3 shadow-around'>
      <div className='mb-1 text-white font-bold select-none'>Player Colors</div>

      <TwRow>
        {playersInRoom
          .filter((p) => !!p.color)
          .sort((a, b) => a.number - b.number)
          .map(({ color, uid, hinter, name, isMyTurn }) => {
            const isCurrentPlayer = currentPlayer.uid === uid
            const bgHinter = hinter ? 'border-slate-500 bg-slate-600' : ''
            const bgMyTurn = isMyTurn ? 'bg-slate-400' : 'bg-transparent'

            const boxClassName = [
              'flex flex-col items-center text-white font-bold border',
              bgHinter,
            ].join(' ')

            return (
              <div key={`player-color-${color}`} className={boxClassName}>
                <CellClickable className={bgMyTurn}>
                  <MarkerLG className={color}>{isCurrentPlayer && 'U'}</MarkerLG>
                </CellClickable>

                {name}
              </div>
            )
          })}
      </TwRow>
    </div>
  )
}
