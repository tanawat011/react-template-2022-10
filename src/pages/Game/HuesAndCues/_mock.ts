import type { RoomPlayer } from './_type'

const PLAYER_BASE: RoomPlayer = {
  cells: [],
  uid: '1',
  name: 'Player 1',
  score: 0,
  color: '',
  number: 1,
}

const result = 'F4'
const choice = ['F4', 'A9', 'P27', 'E15']

export const MOCK: {
  ONE_PLAYER: RoomPlayer[]
  TWO_PLAYER: RoomPlayer[]
  TWO_PLAYER_LAST_IS_HINTER: RoomPlayer[]
  THREE_PLAYER: RoomPlayer[]
  THREE_PLAYER_SECOND_IS_HINTER: RoomPlayer[]
  THREE_PLAYER_LAST_IS_HINTER: RoomPlayer[]
  THREE_PLAYER_LAST_IS_TURN: RoomPlayer[]
  HINTER_NO_CHOICE: RoomPlayer[]
  HINTER_NO_SELECT_CHOICE: RoomPlayer[]
  HINTER_SELECTED_CHOICE: RoomPlayer[]
  PLAYER_NO_COLOR: RoomPlayer[]
  PLAYER_HAS_COLOR: RoomPlayer[]
  PLAYER_HAS_COLOR_START_GAME: RoomPlayer[]
  PLAYER_HAS_COLOR_START_GAME_SUBMIT_RESULT: RoomPlayer[]
} = {
  ONE_PLAYER: [
    {
      ...PLAYER_BASE,
      cells: ['D17'],
      number: 1,
      hinter: true,
      isMyTurn: true,
      color: 'bg-[#0758a5]',
    },
  ],
  TWO_PLAYER: [
    {
      ...PLAYER_BASE,
      number: 1,
      hinter: true,
    },
    {
      ...PLAYER_BASE,
      number: 2,
      isMyTurn: true,
    },
  ],
  TWO_PLAYER_LAST_IS_HINTER: [
    {
      ...PLAYER_BASE,
      number: 1,
      isMyTurn: true,
    },
    {
      ...PLAYER_BASE,
      number: 2,
      hinter: true,
    },
  ],
  THREE_PLAYER: [
    {
      ...PLAYER_BASE,
      number: 1,
      hinter: true,
      result,
      choice,
      owner: true,
      score: 1,
    },
    {
      ...PLAYER_BASE,
      uid: '2',
      number: 2,
      isMyTurn: true,
    },
    {
      ...PLAYER_BASE,
      uid: '3',
      number: 3,
    },
  ],
  THREE_PLAYER_LAST_IS_TURN: [
    {
      ...PLAYER_BASE,
      number: 1,
      hinter: true,
      result,
      choice,
    },
    {
      ...PLAYER_BASE,
      number: 2,
    },
    {
      ...PLAYER_BASE,
      number: 3,
      isMyTurn: true,
    },
  ],
  THREE_PLAYER_SECOND_IS_HINTER: [
    {
      ...PLAYER_BASE,
      number: 1,
      isMyTurn: true,
    },
    {
      ...PLAYER_BASE,
      number: 2,
      hinter: true,
    },
    {
      ...PLAYER_BASE,
      number: 3,
    },
  ],
  THREE_PLAYER_LAST_IS_HINTER: [
    {
      ...PLAYER_BASE,
      number: 1,
    },
    {
      ...PLAYER_BASE,
      number: 2,
      isMyTurn: true,
    },
    {
      ...PLAYER_BASE,
      number: 3,
      hinter: true,
    },
  ],
  HINTER_NO_CHOICE: [
    {
      ...PLAYER_BASE,
      number: 1,
      hinter: true,
    },
    {
      ...PLAYER_BASE,
      number: 2,
      isMyTurn: true,
    },
    {
      ...PLAYER_BASE,
      number: 3,
    },
  ],
  HINTER_NO_SELECT_CHOICE: [
    {
      ...PLAYER_BASE,
      number: 1,
      hinter: true,
      choice,
    },
    {
      ...PLAYER_BASE,
      number: 2,
      isMyTurn: true,
    },
    {
      ...PLAYER_BASE,
      number: 3,
    },
  ],
  HINTER_SELECTED_CHOICE: [
    {
      ...PLAYER_BASE,
      number: 1,
      hinter: true,
      choice,
      result,
    },
    {
      ...PLAYER_BASE,
      number: 2,
      isMyTurn: true,
    },
    {
      ...PLAYER_BASE,
      number: 3,
    },
  ],
  PLAYER_NO_COLOR: [
    {
      ...PLAYER_BASE,
      number: 1,
      hinter: true,
    },
    {
      ...PLAYER_BASE,
      number: 2,
      isMyTurn: true,
    },
    {
      ...PLAYER_BASE,
      number: 3,
    },
  ],
  PLAYER_HAS_COLOR: [
    {
      ...PLAYER_BASE,
      number: 1,
      hinter: true,
      color: 'bg-[#0758a5]',
      owner: true,
    },
    {
      ...PLAYER_BASE,
      number: 2,
      isMyTurn: true,
      color: 'bg-[#117822]',
    },
    {
      ...PLAYER_BASE,
      number: 3,
    },
  ],
  PLAYER_HAS_COLOR_START_GAME: [
    {
      ...PLAYER_BASE,
      number: 1,
      hinter: true,
      color: 'bg-[#0758a5]',
      owner: true,
      isStartGame: true,
      result: 'F4',
    },
    {
      ...PLAYER_BASE,
      number: 2,
      isMyTurn: true,
      cells: ['F4'],
    },
    {
      ...PLAYER_BASE,
      number: 3,
      cells: ['F5', 'F6'],
    },
  ],
  PLAYER_HAS_COLOR_START_GAME_SUBMIT_RESULT: [
    {
      ...PLAYER_BASE,
      number: 1,
      hinter: true,
      color: 'bg-[#0758a5]',
      owner: true,
      isStartGame: true,
      isSubmitResult: true,
      countRound: 1,
    },
    {
      ...PLAYER_BASE,
      number: 2,
      isMyTurn: true,
    },
    {
      ...PLAYER_BASE,
      number: 3,
    },
  ],
}
