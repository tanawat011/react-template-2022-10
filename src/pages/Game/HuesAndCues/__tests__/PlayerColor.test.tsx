import { renderWithProviders } from 'helpers/test'

import { PlayerColor } from '../PlayerColor'
import { MOCK } from '../_mock'

describe('<PlayerColor />', () => {
  test('match snapshot', () => {
    const updateCurrentPlayer = jest.fn()

    const { asFragment } = renderWithProviders(
      <PlayerColor
        playersInRoom={MOCK.PLAYER_HAS_COLOR}
        currentPlayer={MOCK.PLAYER_HAS_COLOR[0]}
        updateCurrentPlayer={updateCurrentPlayer}
      />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('all players no color', () => {
    const updateCurrentPlayer = jest.fn()

    const { asFragment } = renderWithProviders(
      <PlayerColor
        playersInRoom={MOCK.PLAYER_HAS_COLOR}
        currentPlayer={MOCK.PLAYER_HAS_COLOR[0]}
        updateCurrentPlayer={updateCurrentPlayer}
      />,
    )
    expect(asFragment()).toBeTruthy()
  })
})
