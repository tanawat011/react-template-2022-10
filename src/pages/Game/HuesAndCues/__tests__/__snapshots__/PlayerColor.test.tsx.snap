// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerColor /> match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  margin: 0.25rem;
  width: 1.75rem;
  height: 1.75rem;
  --tw-text-opacity: 1;
  color: rgb(255 255 255 / var(--tw-text-opacity));
  font-weight: 700;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
}

.c2 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.75rem;
  line-height: 1rem;
}

.c3 {
  width: 0.75rem;
  height: 0.75rem;
  border-radius: 9999px;
  border-width: 1px;
  --tw-shadow: 0 25px 50px -12px rgb(0 0 0 / 0.25);
  --tw-shadow-colored: 0 25px 50px -12px var(--tw-shadow-color);
  box-shadow: var(--tw-ring-offset-shadow,0 0 #0000),var(--tw-ring-shadow,0 0 #0000),var(--tw-shadow);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  width: 1.5rem;
  height: 1.5rem;
}

<div
    class="mb-2 p-3 shadow-around"
  >
    <div
      class="mb-1 text-white font-bold select-none"
    >
      Player Colors
    </div>
    <div
      class="c0"
    >
      <div
        class="flex flex-col items-center text-white font-bold border border-slate-500 bg-slate-600"
      >
        <div
          class="c1 c2 bg-transparent"
        >
          <div
            class="c3 c4 bg-[#0758a5]"
          >
            U
          </div>
        </div>
        Player 1
      </div>
      <div
        class="flex flex-col items-center text-white font-bold border "
      >
        <div
          class="c1 c2 bg-slate-400"
        >
          <div
            class="c3 c4 bg-[#117822]"
          >
            U
          </div>
        </div>
        Player 1
      </div>
    </div>
  </div>
</DocumentFragment>
`;
